astropy/units/quantity_helper/function_helpers.py

```diff
@@ -497,6 +501,63 @@ def arange_impl(*args, start=None, stop=None, step=None, dtype=None, device=None
     return np.arange(**kwargs), out_unit, None
 
 
+if NUMPY_LT_2_0:
+    @dispatched_function
+    def empty(shape, dtype=None, order='C'):
+        return empty_impl(shape, dtype, order)
+else:
+    @dispatched_function
+    def empty(shape, dtype=None, order='C', *, device=None):
+        return empty_impl(shape, dtype, order, device=device)
+
+def empty_impl(shape, dtype=None, order='C', *, device=None):
```

Comment:
This implementation function is super short and still has to distinguish between numpy versions - just put them directly in the functions above?